File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/inport/command/StartTourCommand.java
package com.dominikgaller.alpinebooking.booking.core.inport.command;

import java.time.Instant;
import java.util.Optional;

/**
 * Input data carrier for the UC05 – StartTour use case.
 *
 * <p>If {@code startedAt} is absent, the application service resolves it
 * to the current time via {@code ClockPort}.
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}, section 2.
 */
public record StartTourCommand(String guideTourId, Optional<Instant> startedAt) {
}

File: src/main/kotlin/com/example/contractmanagement/guide/core/inport/command/StartTourCommand.kt
package com.example.contractmanagement.guide.core.inport.command

import java.time.Instant

/**
 * Input for the StartTour use case (UC05).
 *
 * @property startedAt the actual start time. Null means the caller has no authoritative
 *   time and the driver reads `ClockPort` instead. A guide pressing "start" now sends
 *   nothing; a client replaying a recorded start sends the recorded instant.
 *
 * SDD: see `documentation/ports/start-tour.inport.spec.md` § 2.1.
 */
data class StartTourCommand(
    val guideTourId: String,
    val startedAt: Instant?,
)
